sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyClient.java

```diff
@@ -0,0 +1,227 @@
+import com.azure.security.keyvault.keys.models.Key;
+
+import static com.azure.core.implementation.util.FluxUtil.withContext;
+
+@ServiceClient(builder = KeyClientBuilder.class, isAsync = true, serviceInterfaces = CryptographyService.class)
+public final class CryptographyClient {
+    private CryptographyAsyncClient client;
+
+    /**
+     * Creates a KeyClient that uses {@code pipeline} to service requests
+     *
+     * @param client The {@link CryptographyAsyncClient} that the client routes its request through.
+     */
+    CryptographyClient(CryptographyAsyncClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Gets the public part of the configured key. The get key operation is applicable to all key types and it requires the {@code keys/get} permission.
+     *
+     * @throws ResourceNotFoundException when the configured key doesn't exist in the key vault.
+     * @return A {@link Response} whose {@link Response#value() value} contains the requested {@link Key key}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<Key> getKey() {
+        return withContext(context -> client.getKey(context)).block();
+    }
+
+    /**
+     * Encrypts an arbitrary sequence of bytes using the configured key. Note that the encrypt operation only supports a
+     * single block of data, the size of which is dependent on the target key and the encryption algorithm to be used. The encrypt
+     * operation is supported for both symmetric keys and asymmetric keys. In case of asymmetric keys public portion of the key is used
+     * for encryption. This operation requires the keys/encrypt permission.
+     *
+     * <p>The {@link EncryptionAlgorithm encryption algorithm} indicates the type of algorithm to use for decrypting the specified encrypted content. Possible values
+     * for assymetric keys include: {@link EncryptionAlgorithm#RSA1_5 RSA1_5}, {@link EncryptionAlgorithm#RSA_OAEP RSA_OAEP} and {@link EncryptionAlgorithm#RSA_OAEP_256 RSA_OAEP_256}.
+     * Possible values for symmetric keys include: {@link EncryptionAlgorithm#A128CBC A128CBC}, {@link EncryptionAlgorithm#A128CBC_HS256 A128CBC-HS256},
+     * {@link EncryptionAlgorithm#A192CBC A192CBC}, {@link EncryptionAlgorithm#A192CBC_HS384 A192CBC-HS384}, {@link EncryptionAlgorithm#A256CBC A256CBC} and {@link EncryptionAlgorithm#A256CBC_HS512 A256CBC-HS512} </p>
+     *
+     * @param algorithm The algorithm to be used for encryption.
+     * @param plaintext The content to be encrypted.
+     * @throws ResourceNotFoundException if the key cannot be found for encryption.
+     * @return A {@link EncryptResult} whose {@link EncryptResult#cipherText() cipher text} contains the encrypted content.
+     */
+    public EncryptResult encrypt(EncryptionAlgorithm algorithm, byte[] plaintext, byte[] iv, byte[] authenticationData) {
+        return withContext(context -> client.encrypt(algorithm, plaintext, context, iv, authenticationData)).block();
```

Comment:
Sync clients cannot use `withContext`. You need to explicitly add a `Context` to the API args. Note that `Context` is supported only for APIs that return `Response<T>`.

Reply:
I cannot have Response <T> return types except for getKey operation.
Should I avoid having Context overloads ?